### ISSUE TITLE
Need support "not included" relationship item

### ISSUE THREAD

**Issue by deden** [infinum/Japx#18]
## Expected Behavior
If a JSON with relationships already determined, but the related item not stored in `included`, it will be decoded as null.
Is it the correct behavior?

```
{
    "data": [
        {
            "id": "33",
            "type": "time_off_request",
            "attributes": {
                "status": "approved",
                "start_date": "2019-01-14",
                "end_date": "2019-01-31"
            },
            "relationships": {
                "user": {
                    "data": {
                        "id": "25",
                        "type": "user"
                    }
                },
                "policy": {
                    "data": {
                        "id": "24",
                        "type": "time_off_policy"
                    }
                }
            }
        }
    ],
    "included": [
        {
            "id": "25",
            "type": "user",
            "attributes": {
                "id": 20550,
                "name": "user name",
                "email": "user@email.com",
                "avatar": "",
                "avatar_url": null
            }
        }
    ],
    "meta": {
        "total_pages": 1,
        "total_count": 1
    },
    "links": {
        "self": "url link"
    }
}
```

Decoded as :

```
{
  "links" : {
    "self" : "url link"
  },
  "meta" : {
    "total_pages" : 1,
    "total_count" : 1
  },
  "data" : [
    {
      "status" : "approved",
      "policy" : null,
      "id" : "33",
      "start_date" : "2019-01-14",
      "type" : "time_off_request",
      "end_date" : "2019-01-31",
      "user" : {
        "avatar" : "",
        "id" : 20550,
        "email" : "user@email.com",
        "type" : "user",
        "avatar_url" : null,
        "name" : "user name"
      }
    }
  ]
}
```

## Actual Behavior
I think the related relationship item (_policy_ field with type `time_off_policy`) not decoded as null, i.e :

```
{
  "links" : {
    "self" : "url link"
  },
  "meta" : {
    "total_pages" : 1,
    "total_count" : 1
  },
  "data" : [
    {
      "status" : "approved",
      "policy" : {
                        "id": "24",
                        "type": "time_off_policy"
                    },
      "id" : "33",
      "start_date" : "2019-01-14",
      "type" : "time_off_request",
      "end_date" : "2019-01-31",
      "user" : {
        "avatar" : "",
        "id" : 20550,
        "email" : "user@email.com",
        "type" : "user",
        "avatar_url" : null,
        "name" : "user name"
      }
    }
  ]
}
```

What do you think? Is it the correct implementation?
Or, is there a way to force the library to decode it as above?

Thanks

**Comment by iambatoan** [comment]
same issue

**Comment by morinpic** [comment]
same issue

**Comment by morinpic** [comment]
same issue

**Comment by zahraasg** [comment]
same issue

**Comment by hainhathoang88** [comment]
Same issue.

**Comment by Truba** [comment]
Hi Guys, my apologies for waiting on this. 
We are adding `JapxDecodingOptions` with property `parseNotIncludedRelationships` that will fix just this.

I'll also later on be adding something like:
```
enum JapxRelationship<T>: Codable {
    case pointer(type: String, id: String)
    case element(T)
}
```
To the codable extension to encourage type safety while using this.

**Comment by fgulan** [comment]
This is released with 2.2.0 @hainhathoang88 @iambatoan @deden, thanks for the input!
